src/datasets/data_files.py

```diff
@@ -134,6 +144,23 @@ def sanitize_patterns(patterns: Union[Dict, List, str]) -> Dict[str, Union[List[
         return sanitize_patterns(list(patterns))
 
 
+def _expand_globstar(pattern: str) -> str:
```

Comment:
I think we should fully align with the new fsspec logic, so ideally it would be great to remove the extra `_expand_globstar` logic.

Therefore it's fine to simply remove the test examples using unsupported patterns like `data/**`, `**train*` or `**test*`, and and update the remaining patterns that still use `**` like `DEFAULT_PATTERNS_ALL` and `METADATA_PATTERNS` (and some patterns from the new `KEYWORDS_IN_PATH_NAME_BASE_PATTERNS ` you added)

What do you think ?

Reply:
This is fine with me, I was just going off of the original comment from https://github.com/huggingface/datasets/issues/6644 which said the intention was to continue supporting the old glob behavior

Reply:
cc @albertvillanova IMO we can drop the `<prefix>**<suffix>` support, since it's been a behavior that only existed in `fsspec` and no other well known globbing tool afaik

I double checked and actually we just need to remove the test involving `**train*` and `**test*`.

No need to modify `data/**`, `DEFAULT_PATTERNS_ALL`, `METADATA_PATTERNS` or `KEYWORDS_IN_PATH_NAME_BASE_PATTERNS` as I said in my previous message.

Reply:
@lhoestq I've pushed an update that removes the unsupported `**train*`/`**test*` tests, removes the `_expand_globstar` behavior, and simplifies the updated keyword patterns to
```
    KEYWORDS_IN_PATH_NAME_BASE_PATTERNS = [
        "**/{keyword}[{sep}]*",
        "**/{keyword}/**",
        "**/*[{sep}]{keyword}[{sep}]*",
        "**/*[{sep}]{keyword}[{sep}]*/**",
    ]
```